lnrpc/invoicesrpc/addinvoice.go

```diff
@@ -370,6 +370,15 @@ func AddInvoice(ctx context.Context, cfg *AddInvoiceConfig,
 	)
 	options = append(options, zpay32.Features(invoiceFeatures))
 
+	// Generate and set a random payment address for this invoice. If the
+	// sender understands payment addresses, this can be used to avoid
+	// intermediaries probing the receiver.
+	var paymentAddr [32]byte
+	if _, err := rand.Read(paymentAddr[:]); err != nil {
+		return nil, nil, err
+	}
+	options = append(options, zpay32.PaymentAddr(&paymentAddr))
```

Comment:
Don't we also need to write this to disk, so we can compare it against what we receive from the extended HTLC?

Reply:
Scratch that, see it below in the expanded diff.